openslides/motions/static/js/motions/site.js

```diff
@@ -1035,18 +1039,40 @@ angular.module('OpenSlidesApp.motions.site', [
                 });
             }
         };
-        // delete selected motions
-        $scope.deleteMultiple = function () {
+        var selectModeAction = function (predicate) {
             angular.forEach($scope.motionsFiltered, function (motion) {
-                if (motion.selected)
-                    Motion.destroy(motion.id);
+                if (motion.selected) {
+                    predicate(motion);
+                }
             });
             $scope.isSelectMode = false;
             $scope.uncheckAll();
         };
-        // delete single motion
-        $scope.delete = function (motion) {
-            Motion.destroy(motion.id);
+        // delete selected motions
+        $scope.deleteMultiple = function () {
+            selectModeAction(function (motion) {
+                $scope.delete(motion);
+            });
+        };
+        // set status for selected motions
+        $scope.setStatusMultiple = function (stateId) {
+            selectModeAction(function (motion) {
+                $scope.updateState(motion, stateId);
+            });
+        };
+        // set category for selected motions
+        $scope.setCategoryMultiple = function (categoryId) {
+            selectModeAction(function (motion) {
+                motion.category_id = categoryId === 'none' ? null : categoryId;
```

Comment:
This looks strange. In which cases has the categoryId the string `'none'` instead of `null` or `undefined`?